src/main/java/stsjorbsmod/powers/SacrificePower.java

```diff
@@ -0,0 +1,63 @@
+package stsjorbsmod.powers;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.GainEnergyAction;
+import com.megacrit.cardcrawl.actions.common.RemoveSpecificPowerAction;
+import com.megacrit.cardcrawl.actions.utility.UseCardAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.powers.AbstractPower;
+import stsjorbsmod.actions.DecreaseMaxHpAction;
+
+public class SacrificePower extends CustomJorbsModPower {
+    public static final StaticPowerInfo STATIC = StaticPowerInfo.Load(SacrificePower.class);
+    public static final String POWER_ID = STATIC.ID;
+
+    public SacrificePower(AbstractPlayer owner, int stacks) {
+        super(STATIC);
+        this.isTurnBased = true;
+        this.amount = stacks;
+        this.owner = owner;
+        this.updateDescription();
+    }
+
+    @Override
+    public void onUseCard(AbstractCard card, UseCardAction action) {
```

Comment:
I think this implementation is missing a few interactions with other effects that also mess with how much cards cost; in particular, `AbstractPlayer.useCard` has a variety of other effects besides just `costForTurn` that would normally make using a card free, but which this implementation is still going to cost max HP for.

I think the two options for dealing with this would be:
* Repeat the logic from `AbstractPlayer.UseCard` L1693-1694 here to allow 0 hpCost, or
* Scrap `onUseCard` and instead implement the power's logic using an `EnergyManager.use` patch

I would actually lean towards the second option for this case, even though it does involve a new patch. The two main reasons I think the patch might be better in this case are:

* It also avoids the problem of needing to do a hacky thing to handle the X cost card case (X cost cards also use `EnergyManager.use` and already have logic for not calling it if they're supposed to be free to play, so I think the patch solution would just work naturally for them)
* It cooperates better with any other mods that might add Corruption-like effects by patching `AbstractPlayer.useCard`

Reply:
If we do the patch, probably best to update the wording on the description to avoid the "to play cards" part, since in theory it would also affect any other cases that cost energy outside of playing a card (I don't think there are currently any such cases in the base game or our mod, but maybe there are other mods that add them)